### PR TITLE
Feature/5646 super admin slack notifications

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def with_current_user_and_team(user = nil, team = nil)
 
   def setup
     Pusher::Client.any_instance.stubs(:trigger)
-    CheckNotifications::Slack::Request.any_instance.stubs(:request).returns(nil)
+    # CheckNotifications::Slack::Request.any_instance.stubs(:request).returns(nil)
```
Comment (infojunkie) [inline test/test_helper.rb]: you can completely remove this line instead of commenting

Comment (infojunkie) [inline test/test_helper.rb]: and the `unstub` part too of course

File: config/initializers/active_record_extensions.rb [hunk 2]
```diff
@@ -53,6 +52,43 @@ def destroy_annotations_and_versions
...
+    text = text.tr("\n", ' ')
+    "<#{url}|#{text}>"
+  end
+
+  def to_slack_quote(text)
+    text = I18n.t(:blank) if text.blank?
+    text = self.to_slack(text)
+    text.insert(0, "\n") unless text.start_with? "\n"
+    text.gsub("\n", "\n>")
+  end
```
Comment (caiosba) [inline config/initializers/active_record_extensions.rb]: Just a suggestion: I think that methods from line 63 until line 83 could be converted to methods of the Slack bot. They don't use any information of the `ActiveRecord` model that inherits them.

File: test/models/project_test.rb [hunk 3]
```diff
@@ -291,22 +291,6 @@ def setup
...
-    assert_equal 0, CheckNotifications::Slack::Worker.jobs.size
-    with_current_user_and_team(u, t) do
-      p = create_project team: t
-      assert_equal 1, CheckNotifications::Slack::Worker.jobs.size
-      CheckNotifications::Slack::Worker.drain
-      assert_equal 0, CheckNotifications::Slack::Worker.jobs.size
-      Rails.unstub(:env)
-    end
-  end
-
```
Comment (caiosba) [inline test/models/project_test.rb]: If Slack is notified when a project is created, why this test was removed?

File: test/workers/slack_notification_worker_test.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    create_team_user team: t, user: u, role: 'owner'
+    assert_equal 0, SlackNotificationWorker.jobs.size
+    with_current_user_and_team(u, t) do
+      p = create_project team: t
+      assert_equal 1, SlackNotificationWorker.jobs.size
+      SlackNotificationWorker.drain
+      assert_equal 0, SlackNotificationWorker.jobs.size
+      Rails.unstub(:env)
+    end
+  end
```
Comment (caiosba) [inline test/workers/slack_notification_worker_test.rb]: If this test applies to the "project creation" event, ignore my previous note.

**Pull request merged**